pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -220,16 +182,18 @@ func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string, caBundleRevi
 		upstreamResolvers.Policy = dns.Spec.UpstreamResolvers.Policy
 	}
 
-	if transport == operatorv1.TLSTransport {
-		// tls cannot be configured for system resolv config
+	if dns.Spec.UpstreamResolvers.TransportConfig.Transport == operatorv1.TLSTransport {
+		// tls can only be configured for ip addresses
 		for _, upstream := range upstreamResolvers.Upstreams {
 			if upstream.Type == operatorv1.SystemResolveConfType {
-				return nil, errTransportTLSConfiguredForSysResConf
+				return nil, errTransportTLSConfiguredForNonIP
```

Comment:
Hmmm. I think your original `errTransportTLSConfiguredForSysResConf` was really quite nice in this specific error case because it was very descriptive. `errTransportTLSConfiguredForNonIP` isn't wrong here, but it doesn't provide as much help as `errTransportTLSConfiguredForSysResConf`.

What about having both `errTransportTLSConfiguredForSysResConf` and `errTransportTLSConfiguredForNonIP`?